internal/extension/smartagentextension/README.md

```diff
@@ -0,0 +1,24 @@
+# SignalFx Smart Agent Extension
+
+The `smartagent` extension provides a mechanism to specify config options that are not
+just specific to a single instance of the `smartagent` receiver but are applicable to
+all instances of the `smartagent` receiver.
+
+To begin with, this extension will provide a mechanism to specify config options to configure
+collectd. These options are mapped to the [collectd config options](https://docs.signalfx.com/en/latest/integrations/agent/config-schema.html#collectd)
+in the SignalFx Agent. Note that if this extension is not configured, the defaults in `smartagent`
+receiver will be used.
+
+In the below example configuration, `config_dir` and `bundle_dir` will be used for all instances
+of the `smartagent` receiver that wrap around a collectd based monitor.
+
+```yaml
+extensions:
+  smartagent:
+    collectd:
+      config_dir: /etc/collectd/collectd.conf
+      bundle_dir: /opt/bin/collectd/
```

Comment:
I think this should be top level in the extension since it's applicable to more than collectd and not able to be specified in its agent config: https://github.com/signalfx/signalfx-agent/blob/7bed2632de74e3e9359343d624dd6ce8d2452e07/pkg/core/config/config.go#L390

Reply:
That's a good point. I'll move this option to the receiver.

Reply:
Thanks, mind updating the example as well?